articles/compound-component-pattern-101.md

```diff
@@ -0,0 +1,148 @@
+---
+title: "Compound Components In React 101"
+subtitle: "Compound Components In React 101"
+date: "2023-07-21"
+tags: "#React #typescript #designPatterns #fundamentals"
+author : "Mateo Sierra"
+---
+
+The web has evolved and the way we build websites has followed , with the goal of being able to develop bigger and more complex user interfaces(UI), developers started dividing the user interface into smaller fragments called  components, as a result of this there was a need of ways of creating and  composing components in a predictible way.
+
+React's Compound Component Pattern is a powerful technique which allows developers to build flexible and reusable components by combining fragments of UI together. We'll explore what compound components are, why we use them, the advantages they offer, and we'll walk through a simple example which will teach you how to write them and take advantage of this pattern.
+
+## Bringing order to chaos: why use compound components?
+
+It's incredibly easy to write messy code, you start without any kind of structure in mind and you will end up your coding session with working features but, the next day adding new features to your code will feel more and more difficult, and the following days it will only get worse.
+
+Compound components will bring some order to your code, this will ensure your code stays managable. allowing you to fix bugs, add feautures and even refactor it without much struggle on the long run, let's review the advantages
```

Comment:
```suggestion
Compound components will bring some order to your code, this will ensure your code stays managable. allowing you to fix bugs, add feautures and even refactor it without much struggle on the long run, let's review the advantages.
```